src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -695,6 +701,79 @@ class spell_mage_water_elemental_freeze : public SpellScript
     }
 };
 
+// Ice Lance - 30455
+class spell_mage_ice_lance : public SpellScript
+{
+    PrepareSpellScript(spell_mage_ice_lance);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({
+            SPELL_MAGE_ICE_LANCE,
+            SPELL_MAGE_ICE_LANCE_TRIGGER
+        });
+    }
+
+    void HandleOnHit(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+        if (!caster && !target)
+            return;
+
+        // Thermal Void
+        if (caster->HasAura(SPELL_MAGE_THERMAL_VOID) && (target->HasAuraState(AURA_STATE_FROZEN) || caster->HasAura(SPELL_MAGE_FINGERS_OF_FROST)))
+        {
+            if (Aura* icyVeins = caster->GetAura(SPELL_MAGE_ICY_VEINS))
+            {
+                if (SpellInfo const* thermalVoidInfo = sSpellMgr->GetSpellInfo(SPELL_MAGE_THERMAL_VOID, GetCastDifficulty()))
+                    if (SpellEffectInfo const* eff0 = thermalVoidInfo->GetEffect(EFFECT_0))
+                    {
+                        int32 increaseDuration = eff0->CalcValue() * IN_MILLISECONDS;
+                        int32 newDuration = (icyVeins->GetDuration() + increaseDuration);
+                        icyVeins->SetDuration(newDuration);
+                    }
+            }
+        }
+
+        if (Aura* fingersOfFrost = caster->GetAura(SPELL_MAGE_FINGERS_OF_FROST))
+        {
+            uint8 stack = fingersOfFrost->GetStackAmount();
```

Comment:
```cpp
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/Spells/spell_mage.cpp:741:19: fatal error: 
      unused variable 'stack' [-Wunused-variable]
            uint8 stack = fingersOfFrost->GetStackAmount();
                  ^
```
Another case of unused variable (which could be present further down as well, not sure), so you may want to reconsider what can be used in that line.

Reply:
I am not a real coder, but I wonder if you either should drop `uint8 stack` and leave
```cpp
fingersOfFrost->GetStackAmount();
```
or change the line below to be
```cpp
stack->ModStackAmount(-1);
```

Reply:
yes, he can just drop that line